QuizApp/MVP/ViewController/LoginViewController.swift

```diff
@@ -17,19 +17,25 @@ class LoginViewController : UIViewController, LoginViewDelegate {
     private var passwordTextField:InputField!
     private var passwordToggleIconView:UIImageView!
     private var stackView:UIStackView!
-    
-    private var router: AppRouterProtocol!
     private var presenter: LoginPresenter!
     
     convenience init(router: AppRouterProtocol) {
         self.init()
-        self.router = router
-        self.presenter = LoginPresenter(delegate: self)
+        self.presenter = LoginPresenter(delegate: self, router: router)
     }
     
     override func viewDidLoad() {
         super.viewDidLoad()
-        buildViews()
+        // remember login
+        let defaults = UserDefaults.standard
+        guard let _ = defaults.object(forKey: "user_id"),
+              let _ = defaults.object(forKey: "user_token")
+        else{
+            buildViews()
+            return
+        }
+        self.styleViews()
+        self.presenter.presentGoodLogin()
```

Comment:
Why would you call this here? Isn’t this skipping the whole login process?

Reply:
If user is logged in, aplication remembers it and skips login